R/rvar-factor.R

```diff
@@ -0,0 +1,224 @@
+#' Factor random variables of arbitrary dimension
+#'
+#' Random variables backed by [factor]-like arrays of arbitrary dimension.
+#'
+#' @name rvar_factor
+#'
+#' @inheritParams rvar
+#' @inheritDotParams base::factor
+#'
+#' @details
+#'
+#' A subtype of [rvar()] that represents a (possibly multidimensional) sample of
+#' a [factor] or an [ordered] factor. It is otherwise very similar to the basic [rvar()]:
+#' it is backed by a multidimensional array with draws as the first dimension.
+#' The primary difference is that the backing array has class `"factor"` (for [rvar_factor()])
+#' or `c("ordered", "factor")` (for [rvar_ordered()]). If you
+#' pass a [factor] or [ordered] factor to [rvar()] it will automatically return
+#' an object with the classes `"rvar_factor"` or `c("rvar_ordered", "rvar_factor")`.
+#'
+#' See [rvar()] for more details on the internals of the random variable datatype.
+#'
+#' @seealso [as_rvar_factor()] to convert objects to `rvar_factor`s. See [rdo()], [rfun()], and
+#' [rvar_rng()] for higher-level interfaces for creating `rvar`s.
+#'
+#' @return An object of class `"rvar_factor"` representing a `factor`-like random variable.
+#'
+#' @examples
+#'
+#' # TODO: update examples
```

Comment:
Should this be TODO added to the PR before merging?

Reply:
good catch, thanks! done